File: Backend/src/main/java/com/example/project_reactnative/model/entity/Review.java
package com.example.project_reactnative.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
public class Review {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "hotel_id", referencedColumnName = "id")
    private Hotel hotel;

    @ManyToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "room_id", referencedColumnName = "id")
    private Room room;

    private int rating;

    private String comment;

    private LocalDate commentDate = LocalDate.now();

    private LocalDateTime createdAt;
}

File: Frontend/interface/review.ts
import { Hotel } from "./hotel";
import { User } from "./interface";
import { Room } from "./room";

export interface Review {
  id: number;
  hotel: Hotel; 
  user: User; 
  room: Room; 
  rating: number; 
  comment?: string; 
  commentDate: string; 
  createdAt: string;
}

export interface ReviewResponse {
  id: number;
  fullName: string;
  avatar: string;
  comment?: string; 
  commentDate: string; 
  rating: number
}

export interface ReviewRequest{
  hotelId?: number;
  roomId?: number;
  userId?: number;
  rating: number;
  comment: string;
}